client/src/main/java/org/evosuite/ga/metaheuristics/TestSuiteAdapter.java

```diff
@@ -434,13 +447,14 @@ final public void notifyIteration() { // called by the adaptee
     }
 
     @Override
-    final protected void sortPopulation() throws UnsupportedOperationException {
-        throw new UnsupportedOperationException("not implemented");
+    final protected void sortPopulation() {
+        algorithm.sortPopulation();
     }
 
     @Override
     final public List<TestSuiteChromosome> getPopulation() throws UnsupportedOperationException {
-        throw new UnsupportedOperationException("not implemented");
+        // The documentation of GeneticAlgorithm::getPopulation states, that an accessor to the population is returned.
```

Comment:
Just to clarify: we could (in principle) implement `getPopulation()` as follows:
```Java
final public List<TestSuiteChromosome> getPopulation() 
    return Collections.singletonList(algorithm.getPopulation().stream().collect(TestChromosome.toTestSuiteCollector));
}
```
But this solution was not chosen because it violates the contract described in the JavaDoc of `GeneticAlgorithm::getPopulation`:

> `Accessor for population Chromosome`

To me, the JavaDoc is ambiguous. Does it really state that we must return a reference to the actual population and that we can't return a reference to (some) copy of the population?

Reply:
I understood the JavaDoc, that i can add/delete elements by editing the population:

e.g.
`ga.getPopulation().clear(); `

would clear the population of ga.
I was concerned, that some code will break, if the population can not be accessed by getPopulation().

I looked at the usages of getPopulation() and have not found a case, where the Population is edited with the accessor.
Considering this, i would recommend, that we use your proposed Implementation of the function.

Maybe we should edit the JavaDoc of getPopulation() in GeneticAlgorithm aswell and return a copy of the population?

Reply:
>  Considering this, i would recommend, that we use your proposed Implementation of the function.

Great :+1: Could you update the changes in your PR in that respect?

> Maybe we should edit the JavaDoc of getPopulation() in GeneticAlgorithm aswell and return a copy of the population?

I think we should. Your comment made me aware that the current implementation of `getPopulation()`
```Java
  public List<T> getPopulation() {
        return population;
    }
```
breaks encapsulation :frowning: And I agree with you, this is bad and we should fix this since it allows callers to mess with the `GeneticAlgorithm`'s internal state. But instead of returning a copy maybe it also suffices to do this?
```Java
 public List<T> getPopulation() {
        return Collections.unmodifiableList(population);
    }
```
When someone tries to add or delete chromosomes from the returned list they would get an `UnsupportedOperationException` exception. This would be fine for me. Returning a copy of the list would allow callers to modify the list, which with I am also fine since it doesn't affect the internal state of the GA either.

Reply:
I think 50c7bae resolves this conversation.